### PR TITLE
Remove steering wheel offset for planner slow down for curves

### LINKED ISSUE #33667: possible invalid use of steeringAngleDeg without calibration
### Describe the bug

**Issue**
Car struggles to accelerate smoothly when overtaking a car. Car also does not speed up to set speed when not lead visible.

**Attempted Troubleshooting**
Reflash (flash.comma.ai) and reinstall
Turn off radar tracks and run VOACC
Update to latest master-ci
Run Longitudinal Maneuvers Testing Tool

**accel Pulsing**
Screenshot 2024-09-28 at 3 12 50 PM
image

**vEgo not hitting setSpeed**
Screenshot 2024-09-28 at 3 21 54 PM

### Provide a route where the issue occurs

162796f1469f2f1b/00000001--b37821a3c3

### openpilot version

3095fa79279e8c6424571a0195d04f1fcce579fa

### Additional info

**Disclaimer**
My car does not have long support upstream due to using extra panda, but code is still nearly identical to camera cars. Code changes should not have this affect on longitudinal plan (https://github.com/garrettpall/openpilot/commits/gm-long-test/)

### PR DESCRIPTION
**Pull request by twilsonco**
**Description**

In ``selfdrive/controls/lib/longitudinal_planner.py``, the 	`limit_accel_in_turns()` function limits long accel based on lat accel. It uses a back of the hand calculation of lateral accel based on steering angle that does not include a car's steer angle offset (as observed by `liveParams`). In cars with a large steer offset, the `carState` steer angle can be high enough *when traveling in a straight line* that this function ends up limiting long accel. This problem is reported in #33667.

Additionally,  `limit_accel_in_turns()` has a `FIXME` comment: "This function to calculate lateral accel is incorrect and should use the VehicleModel"

This PR implements this `FIXME` change, using a `VehicleModel` updated with `liveParams` values to calculate lateral accel more accurately. It also includes `liveParams` average steer angle offset in order to fix the problem described above.

**Verification**

*Awaiting verification route on this commit*

@garrettpall drove on these changes and verified the problem is resolved and everything drives as it should. However that was not on this specific commit. 

Fixes #33667

### CODE REVIEW

**Comment by sshane** [comment]
While using the VM here is correct, we have avoided making this change as these limits were found via in-car testing a few years ago to be comfortable and we don't want to change them until we decide to spend more time re-tuning them. Can you only include the angle offset fix here? That is more immediately mergable.

File: selfdrive/controls/lib/longitudinal_planner.py [hunk 1]
```diff
@@ -127,19 +129,26 @@ def update(self, sm):
...
-    prev_accel_constraint = not (reset_state or sm['carState'].standstill)
+    prev_accel_constraint = not (reset_state or CS.standstill)
 
     if self.mpc.mode == 'acc':
+      # Update VehicleModel
+      LP = sm['liveParameters']
+      x = max(LP.stiffnessFactor, 0.1)
+      sr = max(LP.steerRatio, 0.1)
+      self.VM.update_params(x, sr)
+      current_curvature = self.VM.calc_curvature(math.radians(CS.steeringAngleDeg - LP.angleOffsetAverageDeg), CS.vEgo, LP.roll)
```
Comment (sshane) [inline selfdrive/controls/lib/longitudinal_planner.py:140]: I believe controlsState has this value, but let's only implement the offset compensation here.

Comment (twilsonco) [reply selfdrive/controls/lib/longitudinal_planner.py]: Ah yes, I forgot it was published there.

**Comment by twilsonco** [comment]
Thanks Shane. I've reverted the use of vehicle model as you suggest. Now it's just the addition of angle offset (and cleanup of the repeated use of `sm['carState']`)

**Comment by jyoung8607** [comment]
> Now it's just the addition of angle offset (and cleanup of the repeated use of `sm['carState']`)

The cleanup aspect is a little debatable, the style in which `submaster` objects are referenced is consistent before this PR and would be inconsistent after.

Could we leave that aside, at least for now? It's a good practice for null-effect refactors/cleanups to be considered separately from functional changes like this.

If we just recalculate the angle, right next to the `v_ego` assignment, and pass it to `limit_accel_in_turns`:

```python
  calibrated_steer_angle = sm['carState'].steeringAngleDeg - sm['liveParameters'].angleOffsetAverageDeg
```

Then this becomes a three-line PR with the nature of the change visually obvious.

**Comment by sshane** [comment]
thanks for the PR! I hope you can see how this diff is much easier to merge immediately vs the original commit.

**Comment by twilsonco** [comment]
Yes. Much cleaner this way. Thanks Shane and Jason!

**Pull request merged**